diag_manager/fms_diag_file_object.F90

```diff
@@ -1096,6 +1141,46 @@ subroutine write_axis_metadata(this, diag_axis)
 
 end subroutine write_axis_metadata
 
+!< @brief Writes the field metadata for the file
+subroutine write_field_metadata(this, diag_field, diag_axis)
+  class(fmsDiagFileContainer_type), intent(inout), target :: this            !< The file object
+  class(fmsDiagField_type)        , intent(inout), target :: diag_field(:)   !<
+  class(fmsDiagAxisContainer_type), intent(in)            :: diag_axis(:)    !< Diag_axis object
+
+  class(FmsNetcdfFile_t),  pointer     :: fileobj        !< The fileobj to write to
+  class(fmsDiagFile_type), pointer     :: diag_file      !< Diag_file object to open
+
+  integer :: i           !< For do loops
+  integer :: j           !< For do loops
```

Comment:
This is a field ID not an index for looping.  It should probably be a pointer.

Reply:
This was replaced with a pointer https://github.com/NOAA-GFDL/FMS/pull/1102/commits/6d7341622dc912f07180a080e30fee9914b95f5e